In [14]:
# Downloads happen on Google's own network, not your home connection — much faster
!wget -O ncmapss.zip "https://phm-datasets.s3.amazonaws.com/NASA/17.+Turbofan+Engine+Degradation+Simulation+Data+Set+2.zip"

--2026-08-30 19:47:25--  https://phm-datasets.s3.amazonaws.com/NASA/17.+Turbofan+Engine+Degradation+Simulation+Data+Set+2.zip
Resolving phm-datasets.s3.amazonaws.com (phm-datasets.s3.amazonaws.com)... 16.15.212.209, 16.15.252.154, 16.15.214.54, ...
Connecting to phm-datasets.s3.amazonaws.com (phm-datasets.s3.amazonaws.com)|16.15.212.209|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 15760443389 (15G) [application/zip]
Saving to: ‘ncmapss.zip’

ncmapss.zip         100%[===================>]  14.68G  78.0MB/s    in 3m 6s   

2026-08-30 19:50:31 (80.9 MB/s) - ‘ncmapss.zip’ saved [15760443389/15760443389]



In [15]:
!unzip -q ncmapss.zip -d ncmapss_extracted
!ls ncmapss_extracted   # confirm the folder structure and find N-CMAPSS_DS03-012.h5 inside

replace ncmapss_extracted/17. Turbofan Engine Degradation Simulation Data Set 2/data_set.zip? [y]es, [n]o, [A]ll, [N]one, [r]ename: n
'17. Turbofan Engine Degradation Simulation Data Set 2'   data_set


The first line unzips the archive quietly (no file-by-file listing) and puts everything into a newly created folder named ncmapss_extracted. The second line then lists what actually landed inside that new folder, which is how you confirmed what unzip produced without having to click around in a file browser.

In [16]:
!unzip -q "ncmapss_extracted/17. Turbofan Engine Degradation Simulation Data Set 2/data_set.zip" -d ncmapss_extracted/data_set
!find ncmapss_extracted -name "*.h5"

replace ncmapss_extracted/data_set/data_set/N-CMAPSS_DS01-005.h5? [y]es, [n]o, [A]ll, [N]one, [r]ename: A
ncmapss_extracted/data_set/data_set/N-CMAPSS_DS02-006.h5
ncmapss_extracted/data_set/data_set/N-CMAPSS_DS08d-010.h5
ncmapss_extracted/data_set/data_set/N-CMAPSS_DS07.h5
ncmapss_extracted/data_set/data_set/N-CMAPSS_DS08c-008.h5
ncmapss_extracted/data_set/data_set/N-CMAPSS_DS06.h5
ncmapss_extracted/data_set/data_set/N-CMAPSS_DS05.h5
ncmapss_extracted/data_set/data_set/N-CMAPSS_DS04.h5
ncmapss_extracted/data_set/data_set/N-CMAPSS_DS08a-009.h5
ncmapss_extracted/data_set/data_set/N-CMAPSS_DS01-005.h5
ncmapss_extracted/data_set/data_set/N-CMAPSS_DS03-012.h5


In [ ]:
!find ncmapss_extracted -name "N-CMAPSS_DS03-012.h5"

In [ ]:
!find ncmapss_extracted -name "*.h5"

In [ ]:
!find ncmapss_extracted -type f

ncmapss_extracted/17. Turbofan Engine Degradation Simulation Data Set 2/data_set.zip


In [1]:
from google.colab import drive
drive.mount('/content/drive')

!cp "ncmapss_extracted/data_set/data_set/N-CMAPSS_DS03-012.h5" "/content/drive/MyDrive/edge-ai-fault-diagnosis-aerospace/data/raw/"

Mounted at /content/drive
cp: cannot stat 'ncmapss_extracted/data_set/data_set/N-CMAPSS_DS03-012.h5': No such file or directory


In [2]:
!ls ncmapss_extracted/data_set/data_set/

ls: cannot access 'ncmapss_extracted/data_set/data_set/': No such file or directory


Re-run the download and both unzip steps, since those are what populate /content/. Run these three cells again, in order, exactly as before:

In [3]:
!wget -O ncmapss.zip "https://phm-datasets.s3.amazonaws.com/NASA/17.+Turbofan+Engine+Degradation+Simulation+Data+Set+2.zip"

--2026-08-29 17:51:24--  https://phm-datasets.s3.amazonaws.com/NASA/17.+Turbofan+Engine+Degradation+Simulation+Data+Set+2.zip
Resolving phm-datasets.s3.amazonaws.com (phm-datasets.s3.amazonaws.com)... 54.231.168.113, 16.15.255.220, 16.182.67.57, ...
Connecting to phm-datasets.s3.amazonaws.com (phm-datasets.s3.amazonaws.com)|54.231.168.113|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 15760443389 (15G) [application/zip]
Saving to: ‘ncmapss.zip’

ncmapss.zip         100%[===================>]  14.68G  51.9MB/s    in 5m 0s   

2026-08-29 17:56:25 (50.1 MB/s) - ‘ncmapss.zip’ saved [15760443389/15760443389]



In [7]:
!unzip -q ncmapss.zip -d ncmapss_extracted

replace ncmapss_extracted/17. Turbofan Engine Degradation Simulation Data Set 2/data_set.zip? [y]es, [n]o, [A]ll, [N]one, [r]ename: N


Open the zip file that was hiding inside the first extracted folder, and put its contents into a new data_set subfolder." This is the step that finally produced the actual .h5 files — DS01, DS03, DS08a, and the rest — at the path ncmapss_extracted/data_set/data_set/N-CMAPSS_DS0X....h5 you confirmed with find afterward.

unzipping the nested data_set.zip (the one that actually contains all the .h5 files) into the ncmapss_extracted folder.

In [8]:
!unzip -q "ncmapss_extracted/17. Turbofan Engine Degradation Simulation Data Set 2/data_set.zip" -d ncmapss_extracted/data_set

These are two cp (copy) commands, each doing the same job for a different file.

cp copies a file from one location to another — same idea as copy-pasting a file in File Explorer, just written as a command.

The first quoted path — "ncmapss_extracted/data_set/data_set/N-CMAPSS_DS01-005.h5" — is the source: the actual file being copied, sitting in Colab's temporary local storage, exactly where the second unzip step put it.

In [9]:
from google.colab import drive
drive.mount('/content/drive')

!cp "ncmapss_extracted/data_set/data_set/N-CMAPSS_DS01-005.h5" "/content/drive/MyDrive/edge-ai-fault-diagnosis-aerospace/data/raw/"
!cp "ncmapss_extracted/data_set/data_set/N-CMAPSS_DS08a-009.h5" "/content/drive/MyDrive/edge-ai-fault-diagnosis-aerospace/data/raw/"

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


ls -la lists a folder's contents in detail — -l shows size and date modified, -a shows hidden files too.

This line checked the Drive folder where the cp commands copied your files, confirming all three .h5 files landed with real, non-zero sizes (DS01 ~2.87GB, DS03 ~3.69GB, DS08a ~3.24GB) — proof the copies actually worked, since cp itself prints nothing on success

In [4]:
!ls -la "/content/drive/MyDrive/edge-ai-fault-diagnosis-aerospace/data/raw/"

total 9573741
-rw------- 1 root root 2873351432 Aug 29 18:31 N-CMAPSS_DS01-005.h5
-rw------- 1 root root 3693395776 Aug 29 03:08 N-CMAPSS_DS03-012.h5
-rw------- 1 root root 3236762200 Aug 29 18:31 N-CMAPSS_DS08a-009.h5


Below adding addition files due to baseline project
All 10 .h5 files extracted successfully — full listing confirmed. Nice catch, too: there's also

In [9]:
#listing files
!ls ncmapss_extracted/data_set/data_set/

N-CMAPSS_DS01-005.h5
N-CMAPSS_DS02-006.h5
N-CMAPSS_DS03-012.h5
N-CMAPSS_DS04.h5
N-CMAPSS_DS05.h5
N-CMAPSS_DS06.h5
N-CMAPSS_DS07.h5
N-CMAPSS_DS08a-009.h5
N-CMAPSS_DS08c-008.h5
N-CMAPSS_DS08d-010.h5
N-CMAPSS_Example_data_loading_and_exploration.ipynb
Run_to_Failure_Simulation_Under_Real_Flight_Conditions_Dataset.pdf


In [10]:
# copy all 10 files
!cp "ncmapss_extracted/data_set/data_set/N-CMAPSS_DS02-006.h5" "/content/drive/MyDrive/edge-ai-fault-diagnosis-aerospace/data/raw/"
!cp "ncmapss_extracted/data_set/data_set/N-CMAPSS_DS04.h5" "/content/drive/MyDrive/edge-ai-fault-diagnosis-aerospace/data/raw/"
!cp "ncmapss_extracted/data_set/data_set/N-CMAPSS_DS05.h5" "/content/drive/MyDrive/edge-ai-fault-diagnosis-aerospace/data/raw/"
!cp "ncmapss_extracted/data_set/data_set/N-CMAPSS_DS06.h5" "/content/drive/MyDrive/edge-ai-fault-diagnosis-aerospace/data/raw/"
!cp "ncmapss_extracted/data_set/data_set/N-CMAPSS_DS07.h5" "/content/drive/MyDrive/edge-ai-fault-diagnosis-aerospace/data/raw/"
!cp "ncmapss_extracted/data_set/data_set/N-CMAPSS_DS08c-008.h5" "/content/drive/MyDrive/edge-ai-fault-diagnosis-aerospace/data/raw/"

above: That's 9 files in Drive now — not all 10, but that's intentional, not a mistake. Here's the full list currently there:

N-CMAPSS_DS01-005.h5
N-CMAPSS_DS02-006.h5
N-CMAPSS_DS03-012.h5
N-CMAPSS_DS04.h5
N-CMAPSS_DS05.h5
N-CMAPSS_DS06.h5
N-CMAPSS_DS07.h5
N-CMAPSS_DS08a-009.h5
N-CMAPSS_DS08c-008.h5

That's exactly the 9 files that appear in Akindoju's fault-code mapping table (Table 4.1) — every file his methodology accounts for. The 10th file extracted in Colab, N-CMAPSS_DS08d-010.h5, was deliberately left out of the cp commands, since it isn't in his table and we didn't want to copy and eventually label it based on a guess.

In [11]:
# show all classes
!ls -la "/content/drive/MyDrive/edge-ai-fault-diagnosis-aerospace/data/raw/"

total 25666599
-rw------- 1 root root 2873351432 Aug 29 18:31 N-CMAPSS_DS01-005.h5
-rw------- 1 root root 2450472504 Aug 30 18:49 N-CMAPSS_DS02-006.h5
-rw------- 1 root root 3693395776 Aug 29 03:08 N-CMAPSS_DS03-012.h5
-rw------- 1 root root 3752493952 Aug 30 18:50 N-CMAPSS_DS04.h5
-rw------- 1 root root 2599166216 Aug 30 18:50 N-CMAPSS_DS05.h5
-rw------- 1 root root 2549159720 Aug 30 18:51 N-CMAPSS_DS06.h5
-rw------- 1 root root 2714714776 Aug 30 18:51 N-CMAPSS_DS07.h5
-rw------- 1 root root 3236762200 Aug 29 18:31 N-CMAPSS_DS08a-009.h5
-rw------- 1 root root 2413078176 Aug 30 18:52 N-CMAPSS_DS08c-008.h5


In [12]:
#downloded final 10th file
!cp "ncmapss_extracted/data_set/data_set/N-CMAPSS_DS08d-010.h5" "/content/drive/MyDrive/edge-ai-fault-diagnosis-aerospace/data/raw/"

In [13]:
!ls -la "/content/drive/MyDrive/edge-ai-fault-diagnosis-aerospace/data/raw/"

total 28484016
-rw------- 1 root root 2873351432 Aug 29 18:31 N-CMAPSS_DS01-005.h5
-rw------- 1 root root 2450472504 Aug 30 18:49 N-CMAPSS_DS02-006.h5
-rw------- 1 root root 3693395776 Aug 29 03:08 N-CMAPSS_DS03-012.h5
-rw------- 1 root root 3752493952 Aug 30 18:50 N-CMAPSS_DS04.h5
-rw------- 1 root root 2599166216 Aug 30 18:50 N-CMAPSS_DS05.h5
-rw------- 1 root root 2549159720 Aug 30 18:51 N-CMAPSS_DS06.h5
-rw------- 1 root root 2714714776 Aug 30 18:51 N-CMAPSS_DS07.h5
-rw------- 1 root root 3236762200 Aug 29 18:31 N-CMAPSS_DS08a-009.h5
-rw------- 1 root root 2413078176 Aug 30 18:52 N-CMAPSS_DS08c-008.h5
-rw------- 1 root root 2885034848 Aug 30 19:22 N-CMAPSS_DS08d-010.h5


In [17]:
#re-copy just that one file into Drive to overwrite the truncated version. Run this in a new cell:
!cp "ncmapss_extracted/data_set/data_set/N-CMAPSS_DS08d-010.h5" "/content/drive/MyDrive/edge-ai-fault-diagnosis-aerospace/data/raw/"

In [18]:
# verify the sizes match the local copy and drive copy
!ls -la ncmapss_extracted/data_set/data_set/N-CMAPSS_DS08d-010.h5
!ls -la "/content/drive/MyDrive/edge-ai-fault-diagnosis-aerospace/data/raw/N-CMAPSS_DS08d-010.h5"

-rw-r--r-- 1 root root 2885034848 Jan 13  2021 ncmapss_extracted/data_set/data_set/N-CMAPSS_DS08d-010.h5
-rw------- 1 root root 2885034848 Aug 30 20:05 /content/drive/MyDrive/edge-ai-fault-diagnosis-aerospace/data/raw/N-CMAPSS_DS08d-010.h5


 Both files show exactly `2885034848` bytes — a perfect match. The Drive copy of DS08d is complete now, no more truncation.

Head back to the `01_eda.ipynb` tab and re-run the verification loop cell (the one that checks all 7 new files with h5py). It should now sail through DS08d along with the other 6, printing its shapes, unit list, cycle range, and `hs` values just like the rest. Send a screenshot once it completes.

below: Let's try extracting DS08d with Python's zipfile module instead, which handles large files differently and should sidestep that issue. Run this in a new cell in the Uploaded_NCMAPSS_dataset_to_google_drive notebook:

In [19]:
import zipfile

zip_path = "ncmapss_extracted/17. Turbofan Engine Degradation Simulation Data Set 2/data_set.zip"

with zipfile.ZipFile(zip_path) as z:
    # find the exact entry name inside the archive
    matches = [n for n in z.namelist() if "DS08d" in n]
    print("Found in archive:", matches)

    for name in matches:
        info = z.getinfo(name)
        print(name, "— expected uncompressed size:", info.file_size)

    # extract just this one file fresh, using Python's own unzip logic
    z.extract(matches[0], path="ncmapss_extracted_py")

    # verify its CRC checksum against what the archive expects
    bad_file = z.testzip()
    print("Corrupt entry (None = all good):", bad_file)

Found in archive: ['data_set/N-CMAPSS_DS08d-010.h5']
data_set/N-CMAPSS_DS08d-010.h5 — expected uncompressed size: 2885034848
Corrupt entry (None = all good): None


In [ ]:
#Delete the broken copy from Drive (/data/raw/N-CMAPSS_DS08d-010.h5) so it doesn't get mistakenly used in training later
!rm "/content/drive/MyDrive/edge-ai-fault-diagnosis-aerospace/data/raw/N-CMAPSS_DS08d-010.h5"

Below:Confirmed — the broken DS08d file is gone, and your `data/raw` folder now holds exactly the 9 clean, fully-verified files: DS01, DS02, DS03, DS04, DS05, DS06, DS07, DS08a, and DS08c. That's the complete set matching Akindoju's documented scope.

That closes out data acquisition. Next up on the checklist: decoding `T_var` and `X_v_var` in `01_eda.ipynb`, which has been outstanding since the original EDA pass and is now relevant across all 9 files. Want to tackle that next?

In [22]:
!ls -la "/content/drive/MyDrive/edge-ai-fault-diagnosis-aerospace/data/raw/"

total 25666599
-rw------- 1 root root 2873351432 Aug 29 18:31 N-CMAPSS_DS01-005.h5
-rw------- 1 root root 2450472504 Aug 30 18:49 N-CMAPSS_DS02-006.h5
-rw------- 1 root root 3693395776 Aug 29 03:08 N-CMAPSS_DS03-012.h5
-rw------- 1 root root 3752493952 Aug 30 18:50 N-CMAPSS_DS04.h5
-rw------- 1 root root 2599166216 Aug 30 18:50 N-CMAPSS_DS05.h5
-rw------- 1 root root 2549159720 Aug 30 18:51 N-CMAPSS_DS06.h5
-rw------- 1 root root 2714714776 Aug 30 18:51 N-CMAPSS_DS07.h5
-rw------- 1 root root 3236762200 Aug 29 18:31 N-CMAPSS_DS08a-009.h5
-rw------- 1 root root 2413078176 Aug 30 18:52 N-CMAPSS_DS08c-008.h5
